agent/agent-tooling/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/quickpulse/QuickPulseDataCollector.java

```diff
@@ -51,7 +50,7 @@
 public enum QuickPulseDataCollector {
   INSTANCE;
 
-  private TelemetryClient telemetryClient;
+  private String instrumentationKey;
```

Comment:
we need to support changing the instrumentation key at runtime for Azure Spring Cloud (cc @heyams)
```suggestion
  private Supplier<String> instrumentationKeySupplier;
```